tabular/src/autogluon/tabular/learner/default_learner.py

```diff
@@ -241,107 +236,68 @@ def general_data_processing(self, X: DataFrame, X_val: DataFrame, X_unlabeled: D
-            X_val = self.cleaner.transform(X_val)
-            if len(X_val) == 0:
-                logger.warning(
-                    "############################################################################################################\n"
-                    "WARNING: All X_val data contained low frequency classes, ignoring X_val and generating from subset of X\n"
-                    "\tYour input validation data or training data labels might be corrupted, please manually inspect them for correctness!"
-                )
-                if self.problem_type in [BINARY, MULTICLASS]:
-                    train_classes = sorted(list(y_uncleaned.unique()))
-                    val_classes = sorted(list(y_val_og.unique()))
-                    logger.warning(f"\tTraining Classes: {train_classes}")
-                    logger.warning(f"\tTuning   Classes: {val_classes}")
-                    logger.warning(f"\tTraining Class Dtype: {y_uncleaned.dtype}")
-                    logger.warning(f"\tTuning   Class Dtype: {y_val_og.dtype}")
-                    missing_classes = [c for c in val_classes if c not in train_classes]
-                    logger.warning(f"\tClasses missing from Training Data: {missing_classes}")
-                logger.warning("############################################################################################################")
-
-                X_val = None
-                y_val = None
-                w_val = None
-                holdout_frac = holdout_frac_og
-            else:
-                X_val, y_val = self.extract_label(X_val)
-                y_val = self.label_cleaner.transform(y_val)
-                X_val = self.set_predefined_weights(X_val, y_val)
-                X_val, w_val = extract_column(X_val, self.sample_weight)
-        else:
-            y_val = None
-            w_val = None
+        X_val, y_val, w_val, holdout_frac = self._apply_cleaner_transform(X_val, y_uncleaned, holdout_frac, holdout_frac_og, "X_val")
+        X_test, y_test, w_test, _ = self._apply_cleaner_transform(X_test, y_uncleaned, holdout_frac, holdout_frac_og, "X_test")
 
         self._original_features = list(X.columns)
         # TODO: Move this up to top of data before removing data, this way our feature generator is better
         logger.log(20, f"Using Feature Generators to preprocess the data ...")
-        if X_val is not None:
-            # Do this if working with SKLearn models, otherwise categorical features may perform very badly on the test set
+
+        if X_val is not None or X_test is not None:
             logger.log(
                 15,
-                "Performing general data preprocessing with merged train & validation data, so validation performance may not accurately reflect performance on new test data",
+                "Performing general data preprocessing with merged train & validation data, so validation/test performance may not accurately reflect performance on new test data",
```

Comment:
only log `validation/test` if test is not None. Otherwise it could be confusing to mention `test` in this context when the user didn't specify test data.